accepted/2022/simplify-output-paths.md

```diff
@@ -85,3 +86,10 @@ Before .NET Core, .NET Framework projects generally put their output directly in
 
 This is a classic "pick two of three things" situation, and we believe the least important of the three was trying to preserve `bin\<Configuration>` as an output path.
 
+### NuGet pack logic
+
+Currently, as part of the NuGet pack command, the `_WalkEachTargetPerFramework` target runs the `_GetBuildOutputFilesWithTfm` target for each `TargetFramework`.  This means in this inner build the `TargetFramework` will be a global property, even if there is only one.
+
+This interferes with the logic where the output path depends on whether the `TargetFramework` is specified as a global property.  To fix this, the NuGet pack logic would need to change so that if there is only one target framework, it does not pass the `TargetFramework` global property in this case.
+
+https://github.com/NuGet/Home/issues/12323
```

Comment:
Agreed, NuGet shouldn't pass the TargetFramework in if the project doesn't multi-target. MSBuild honors that in the P2P protocol and undefines TargetFramework in such cases: https://github.com/dotnet/msbuild/blob/704f3497cb93e396fc0c055cc0b17452f2a9e942/src/Tasks/Microsoft.Common.CurrentVersion.targets#L1881-L1887

Reply:
In general, NuGet shouldn't need to do any MSBuild exec invocations in a non multi-targeting build and just gather the package data during the build: https://github.com/NuGet/Home/issues/11530

Reply:
This isn't a P2P call, this is more like an inner/outer build on the same project, where the pack process does an unnecessary inner build even for projects that aren't multi-targeted.  It would be good to remove this, but I also decided it wasn't important to base the pivots on whether `TargetFramework` was a global property, so we don't need the NuGet fix in order to implement the new output paths anymore.